From 31a4fdab9b2d97c485ecb91c43b1e7b1bcbcee86 Mon Sep 17 00:00:00 2001
From: Jakub Tyszkowski <jakub.tyszkowski@tieto.com>
Date: Tue, 18 Nov 2014 12:36:12 +0100
Subject: [PATCH 04/11] sepolicy: Update policies for bluetooth domain apps

This allows bluetooth domain apps to use bluetooth sockets and file
descriptors. This is needed not only for system bluetooth app but also
user apps.

Change-Id: I603d26756de2a5ce7dfc21e2017dff99a03a46ed
---
 bluetoothd.te | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/bluetoothd.te b/bluetoothd.te
index 030f104..d693061 100644
--- a/bluetoothd.te
+++ b/bluetoothd.te
@@ -28,10 +28,19 @@ allow bluetoothd bluetoothd_exec:file execute_no_trans;
 # TODO: minimize this set
 allow bluetoothd self:socket { create_socket_perms accept listen recv_msg send_msg lock recvfrom sendto name_bind relabelfrom relabelto };
 
-# stream sockets
+# Allow clients to use a socket provided by the bluetooth app.
 allow bluetoothd { bluetooth mediaserver }:unix_stream_socket connectto;
-allow bluetooth bluetoothd:unix_stream_socket rw_socket_perms;
+
+# Allow system app to use sockets and fds
 allow bluetooth bluetoothd:fd use;
+allow bluetooth bluetoothd:unix_stream_socket rw_socket_perms;
+
+# Allow user bluetooth apps to use sockets and fds
+allow bluetoothdomain bluetoothd:fd use;
+allow bluetoothdomain bluetoothd:unix_stream_socket { getopt setopt getattr read write ioctl shutdown };
+
+# Other domains that can create and use bluetooth sockets.
+allow bluetoothdomain self:socket create_socket_perms;
 
 #This we might should put to mediaserver.te ?
 allow mediaserver bluetoothd:fd use;
-- 
2.9.3

